backend/modules/experiment.py

```diff
@@ -37,6 +38,7 @@ def __init__(self, session: SessionData):
         self._experimenters = []
         self._participants = {}
         self._logger.info(f"Experiment created: {self}")
+        self.session.creation_time = timestamp()
```

Comment:
Why do you set `creation_time` in the backend? 
I thought it's the responsibility of the frontend:
Start experiment -> frontend sets `creation_time` -> frontend notifies backend.

Same question goes for `start_time` and `end_time`.

Reply:
In general: `creation_time` can be used by the frontend to determine whether the backend is running an experiment for a given session. By default, the frontend does not have this kind of information, since it has no insight into the state of the backend other than the session data. 

From a practical perspective:
-> yes, the frontend could set `creation_time`when starting a session, but:
- If the server shuts down, the frontend cant reset the time
- We would also probably need to synchronize the time between client and server if the client would set the time, otherwise issues regarding time and timezones could lead to unforeseen issues
- If the client would set `creation_time`, `start_time` or `end_time` the server would need to check these fields for faulty values and handle them accordingly. This would need additional error handling

The time also requires no user inputs, so why should it be the responsibility of the client?

I won't go into detail for `start_time` or `end_time`, but similar arguments apply.

Reply:
I imagined it like this:
The frontend sets the `creation_time` variable, and notifies the backend about the new value. The backend then sets the value respectively.

In our current state, the backend sets `creation_time`. Regarding the frontend side, everything's fine, since I will be redirected to the `WatchingRoom` page. But if I go back using the back browser button, I cannot join the experiment, since from the frontend side `creation_time` is not set (but the backend has it already!).

The backend does not update me about the `creation_time`. As soon as I reload the page though, I have the `creation_time` variable. But this functionality is  not complete.
( I cannot use the join button because it is only shown when `creation_time` is greater than 0. It is 0 in the frontend, but > 0 in the backend.)

My current solution is to set the `creation_time` variable separately for the frontend - and then the joining functionality works if I go back using the back browser button, but then we have 2 different joining times - one for the frontend, and one for the backend. And they stay different until the time where I reload or get the `sessionsList` from the backend.

In my opinion, it's not a good functionality as well:
1. In the first case, `creation_time` is only set by the backend -> joining functionality works only after reload.
2. In the second case, `creation_time` is set by both backend and frontend -> different values until frontend reloads.


So we could do 2 possible solutions:
1. Frontend sets `creation_time` and notifies the backend via API request.
2. Backend sets `creation_time` and notifies the frontend via event. (Which would need implementation from your side.)

Reply:
I think the solution to the problem you describe are the update / synchronization messages we already have:
- `CREATED_EXPERIMENT`: send to all experimenters when an experiment is created
- `EXPERIMENT_STARTED`: send to all users in the experiment when an experiment starts
- `EXPERIMENT_ENDED`: send to all users in the experiment when an experiment ends

They currently have an empty data field, but I could add the according timestamp. This would not take too much time from my side.

You have not addressed my concerns / problems with setting the time in the frontend. They still apply for your first solution.

We can also discuss this in our upcoming meeting.

Reply:
Actually utilizing already existing `CREATED_EXPERIMENT`, `EXPERIMENT_STARTED`, and `EXPERIMENT_ENDED` messages would be great for keeping the times synchronized. Then I won't need to set them manually in the frontend, and I think the concerns you addressed would be solved (since backend sets the time and just updates the frontend)
And it would also be nice to get the correct session ID

And yes let's discuss that further in the upcoming meeting as well

Reply:
Meeting summary:
We will use the synchronization messages, I will add the corresponding timestamps to the messages